packages/cozy-doctypes/src/banking/matching-operations.js

```diff
@@ -0,0 +1,94 @@
+const fs = require('fs-extra')
+const groupBy = require('lodash/groupBy')
+
+const getDateOperation = op => op.date.substr(0, 10)
+const groupByDate = ops => groupBy(ops, getDateOperation)
+
+const zipGroup = function *(iterables, grouper) {
+  const grouped = iterables.map(items => groupBy(items, grouper))
+  for (const key of Object.keys(grouped[0]).sort()) {
+    const groups = grouped.map(keyedGroups => keyedGroups[key])
+    yield [key, groups]
+  }
+}
+
+const eitherInclude = (str1, str2) => {
+  return str1.indexOf(str2) > -1 || str2.indexOf(str1)
+}
+
+const differentVendor = (op1, op2) => {
+  return (op1.metadata && op1.metadata.vendor) === (op2.metadata && op2.metadata.vendor)
+}
+
+const sorter = (keyFn, reversed) => (a, b) => {
+  if (keyFn(a) < keyFn(b)) { return reversed ? 1 : -1 }
+  if (keyFn(a) > keyFn(b)) { return reversed ? 1 : -1 }
+  return 0
+}
+
+const byPoints = sorter(x => x.points, true)
+
+const matchOperation = (newOp, existingOps) => {
+  const exactVendorId = existingOps.find(existingOp => existingOp.vendorId === newOp.vendorId)
+  if (exactVendorId) {
+    return { match: exactVendorId, method: 'vendorId' }
+  }
+
+  // Now we try to do it based on label and amount
+  const sorted = existingOps.map(existingOp => {
+    const methods = []
+    const labelPoints = existingOp.originalBankLabel === newOp.originalBankLabel
+      ? (methods.push('originalBankLabel') && 100)
+      : (
+        existingOp.label === newOp.label
+          ? (methods.push('label') && 100) : (
+            eitherInclude(existingOp.label, newOp.label)
+            ? (methods.push('eitherInclude') && 50)
+            : -1000
+        )
+      )
```

Comment:
I think this requires a comment with the explanation of the points system at the beginning of the function.